memphis88/week1/ass2/index.html

```diff
@@ -0,0 +1,106 @@
+<html>
+
+<head>
+    <title>Hello world with react</title>
+    <style type="text/css">
+    body {
+        width: 1000px;
+        margin: auto;
+    }
+    
+    .error {
+        color: red;
+    }
+    </style>
+    <script src="http://fb.me/react-0.14.7.min.js"></script>
+    <script src="http://fb.me/react-dom-0.14.7.min.js"></script>
+    <!-- <script src="http://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.4.4/babel.min.js"></script> -->
+    <script src="http://fb.me/JSXTransformer-0.12.2.js"></script>
+</head>
+
+<body>
+    <!-- 1. Create an inputfield that doesn't accept any numbers -->
+    <!-- 2. Find a way to submit and print the message -->
+    <!-- 3. Validation errors in inputfields should produce an error-message rather than a hello morld message -->
+    <!-- 4. Any error-message should be printed in red color -->
+    <!-- 5. Provide a function that returns all messages ever printed -->
+    <!-- 6. This must be solved by using only native Javascript and react -->
+    <!-- 7. All markup must be produced with react -->
+    <!-- 8. All messages must be printed in the order they where submitted -->
+    <div id="app"></div>
+    <script type="text/jsx">
+    var AssignmentApp = React.createClass({
+    	getInitialState: function() {
+    		return {messages: [], text: '', index: 0};
+    	},
+
+    	onChange: function(e) {
+    		this.setState({ text: e.target.value });
+    	},
+
+    	onKeyDown: function(e) {
+    		if (e.keyCode == 13) { this.onClick() };
+    	},
+
+    	onClick: function() {
+    		var messageObject = {
+    			text: this.state.text,
+    			index: this.state.index
+    		};
+    		var	patt = /^[A-z]+$/;
+			if (!patt.test(this.state.text)) {
+				messageObject.isError = true;
+			}
+    		var newMessages = this.state.messages;
+    		var newIndex = this.state.index;
+    		newIndex++;
+    		newMessages.push(messageObject);
+    		this.setState({messages: newMessages, text: '', index: newIndex});
+    	},
+
+    	showAll: function() {
+    		this.state.messages.map(function(msg) {
```

Comment:
Since you are not transforming the contents of the array, but rather iterate over them, I think `forEach` would be better suited here.